src/Authentication/Authentication.Core/Utilities/PopClient.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.IdentityModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Identity;
+using Azure.Identity.Broker;
+using Microsoft.Identity.Client.NativeInterop;
+
+namespace Microsoft.Graph.PowerShell.Authentication.Core.Utilities
+{
+    public class PopClient
+    {
+        private readonly HttpPipeline _pipeline;
+        private AuthenticationRecord _authenticationRecord;
+        private readonly InteractiveBrowserCredential _interactiveBrowserCredential;
+
+        public PopClient(TokenCredential credential, IAuthContext authContext, ClientOptions options = null)
+        {
+            //_interactiveBrowserCredential = (InteractiveBrowserCredential)credential;
+            _interactiveBrowserCredential = new InteractiveBrowserCredential(new InteractiveBrowserCredentialBrokerOptions(WindowHandleUtlities.GetConsoleOrTerminalWindow()));
+
+            if (!(credential is ISupportsProofOfPossession))
+            {
+                throw new ArgumentException("The provided TokenCredential does not support proof of possession.", nameof(credential));
+            }
+
+            var pipelineOptions = new HttpPipelineOptions(options);
+            pipelineOptions.PerRetryPolicies.Add(new InteractivePopTokenAuthenticationPolicy(_interactiveBrowserCredential, "https://graph.microsoft.com/.default", () => _authenticationRecord));
```

Comment:
What happens if the server doesn't support PoP, it sounds like we fall back to "regular" token.

Reply:
This has been resolved. If the server does not support PoP the request fails.